src/platforms/common/blackpill-f4/blackpill-f4.h

```diff
@@ -115,6 +115,41 @@
 #define SWDIO_PORT TMS_PORT
 #define SWDIO_PIN  TMS_PIN
 
+#if 0
+#define TMS_SET_MODE()     gpio_mode_setup(TMS_PORT, GPIO_MODE_OUTPUT, GPIO_PUPD_NONE, TMS_PIN);
+#define SWDIO_MODE_FLOAT() gpio_mode_setup(SWDIO_PORT, GPIO_MODE_INPUT, GPIO_PUPD_NONE, SWDIO_PIN);
+
+#define SWDIO_MODE_DRIVE() gpio_mode_setup(SWDIO_PORT, GPIO_MODE_OUTPUT, GPIO_PUPD_NONE, SWDIO_PIN);
+
+#else
+
+#define SWDIO_MODE_REG_MULT_PB9 (1 << (9 << 1))
+#define SWDIO_MODE_REG_MULT_PB8 (1 << (8 << 1))
+/* Update when adding more alternative pinouts */
+#define SWDIO_MODE_REG_MULT PINOUT_SWITCH(SWDIO_MODE_REG_MULT_PB9, SWDIO_MODE_REG_MULT_PB8)
+#define SWDIO_MODE_REG      GPIO_MODER(TMS_PORT)
+
+#define TMS_SET_MODE()                                                    \
+	gpio_mode_setup(TMS_PORT, GPIO_MODE_OUTPUT, GPIO_PUPD_NONE, TMS_PIN); \
+	gpio_set_output_options(TMS_PORT, GPIO_OTYPE_PP, GPIO_OSPEED_2MHZ, TMS_PIN);
+
+/* Perform SWDIO bus turnaround faster than a gpio_mode_setup() call */
+#define SWDIO_MODE_FLOAT()                        \
+	do {                                          \
+		uint32_t mode_reg = SWDIO_MODE_REG;       \
+		mode_reg &= ~(0x3 * SWDIO_MODE_REG_MULT); \
```

Comment:
This and the '1' below in SWDIO_MODE_DRIVE() also need `U` suffixing. Might we suggest dropping the `0x` prefix here though for consistency with its counterpart below?